src/orchestrator/unary.rs

```diff
@@ -695,9 +695,10 @@ pub async fn detect(
             response
                 .into_iter()
                 .filter_map(|resp| {
-                    let mut result: TokenClassificationResult = resp.into();
+                    let mut result: TokenClassificationResult = (detector_id.clone(), resp).into();
                     result.start += chunk.offset as u32;
                     result.end += chunk.offset as u32;
```

Comment:
For consistency with how `detector_id` is set on `DetectionResult` and to avoid the weird tuple `From` impl, I would revert the `From` impl change and just update the field like below:

e.g.
```suggestion
                    let mut result: TokenClassificationResult = resp.into();
                    result.detector_id = Some(detector_id.clone());
                    result.start += chunk.offset as u32;
                    result.end += chunk.offset as u32;
```